### PR TITLE
Update `OsuApiService` to fetch solo-scores ("CL detection")

### PR DESCRIPTION
**Pull request by wyvrtn**
As you requested earlier, I've made a small update to `OsuApiService`. The `OsuScore` object has also been rewritten to conform to the new API response from the osu!lazer API.

Associated constructs are listed below:
- [OsuApiService](https://github.com/wyvrtn/huisbot/blob/master/huisbot/Services/OsuApiService.cs)
- [OsuScore](https://github.com/wyvrtn/huisbot/blob/master/huisbot/Models/Osu/OsuScore.cs)

### CODE REVIEW

**Comment by minisbett** [comment]
Could you get rid of any unused properties? There isn't really a need to put more into the model than necessary.

File: huisbot/Services/OsuApiService.cs [hunk 1]
```diff
@@ -200,6 +200,8 @@ public async Task<bool> EnsureAccessTokenAsync()
     try
     {
       // Get the score from the API and check whether a 404 was returned. If so, the score was not found.
+      _http.DefaultRequestHeaders.Remove("x-api-version");
+      _http.DefaultRequestHeaders.Add("x-api-version", "20220705");
```
Comment (minisbett) [inline huisbot/Services/OsuApiService.cs]: This should be implemented in the http client builder in Program.cs instead.

**Review by minisbett — CHANGES_REQUESTED** [review]
Thanks! Could you please get rid of the unused properties in the score model? There's no need for them really.

File: huisbot/Services/OsuApiService.cs (hunk 1 above)
Comment (wyvrtn) [reply huisbot/Services/OsuApiService.cs]: Ok, I'm on it =)

Comment (wyvrtn) [reply huisbot/Services/OsuApiService.cs]: Done.

**Comment by wyvrtn** [comment]
Changes completed. Ready to pull into upstream-master.

**Comment by wyvrtn** [comment]
Ok I'm back. From what I can tell, the API documentation on the new Score object is inaccurate. I've looked at the corresponding TypeScript object in the [osu-web](https://github.com/ppy/osu-web) project, and it seems that the type of the new  `mods` parameter is still `string[]`. Refer to the documentation comment added on the associated `OsuScore` object.

(As of writing my reply I was working on a laptop away from home so I could't retrieve my client details. I also can't remember my osu! account details.)

**Comment by minisbett** [comment]
Huh. Can't you reset your password lol. I'd suggest checking what the API gives you, I'm very sure it does not give you just a string array. The array should be of type `APIMod`, which you can find in the ppy/osu repository.

File: huisbot/Models/Huis/HuisSimulationResponse.cs [hunk 2]
```diff
@@ -149,7 +150,7 @@ public class HuisSimulationScore
     /// The mods of the score, in the osu-tools format.
```
Comment (minisbett) [inline huisbot/Models/Huis/HuisSimulationResponse.cs]:
```suggestion
    /// The mods of the score, in the osu!lazer APIMod format.
```

File: huisbot/Models/Osu/OsuMod.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+using Newtonsoft.Json;
+
+namespace huisbot.Models.Osu;
+
+/// <summary>
+/// Represents an osu!lazer API Mod from the osu! API or Huismetbenen's score simulation.
```
Comment (minisbett) [inline huisbot/Models/Osu/OsuMod.cs]:
```suggestion
/// Represents an osu!lazer API Mod from the osu! API (<see cref="OsuScore"/>) or Huismetbenen score simulation (<see cref="Huis.HuisSimulationResponse"/>).
```

File: huisbot/Models/Osu/OsuMod.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+/// </summary>
+public class OsuMod
+{
+
+  /// <summary>
```
Comment (minisbett) [inline huisbot/Models/Osu/OsuMod.cs]:
```suggestion
  /// <summary>
```

**Review by minisbett — CHANGES_REQUESTED** [review]

**Comment by minisbett** [comment]
Getting the CL mod fetched now works properly. But it seems like the format for hit statistics (100s, 50s, ...) also has been modified in the new model. Would you like to fix that too?

**Comment by wyvrtn** [comment]
All fixed.

**Comment by wyvrtn** [comment]
Anddd... I cleaned up `HuisSimulationResponse`

**Comment by minisbett** [comment]
Looks good now 👍 Thanks alot!

**Pull request merged**